system/jlib/jfile.cpp

```diff
@@ -3626,11 +3627,11 @@ class CLinuxDirectoryIterator : public CDirectoryIterator
     bool next()
     {
         loop {
-            struct dirent dirEntry;
             struct dirent *entry;
             loop {
                 gotst = false;
-                readdir_r(handle, &dirEntry, &entry);
+                CriticalBlock b(sect);
+                entry = readdir(handle);
```

Comment:
Same as the previous CriticalBlock usage.  Please check to make sure it's being used correctly.